samples.cpp

```diff
@@ -0,0 +1,44 @@
+#include "samples.hpp"
+#include <sstream>
+#include <algorithm>
+#include <iterator>
+
+using namespace std;
+
+string RangeData::toString()
+{
+    stringstream strValue;
+    copy(m_range.begin(), m_range.end()-1, ostream_iterator<uint32_t>(strValue, "-"));
+    strValue<<*(m_range.end()-1) << string(", ") << to_string(m_numElements);
+    return strValue.str();
+}
+
+vector<RangeData> findConsecutiveRanges(std::vector<uint32_t> input)
+{
+    uint32_t numElements {1};
+    vector<RangeData> rangeList {};
+
+    if (!input.empty()) {
+        std::sort(input.begin(), input.end());
+        size_t sizeOfcurrentInAmpere = input.size();
+        for(uint32_t counter = 1; counter <= sizeOfcurrentInAmpere; counter++) {
```

Comment:
Please check if the element input[counter] when counter = sizeOfcurrentInAmpere, is not accessing out of range value.
If so then this could be error prone and the behavior undefined, which might need to be handled via a different design